### ISSUE TITLE
Incorrect IP Addresss in logon alerts due to loss of X-Real-IP Header

### ISSUE THREAD

**Issue by danpoltawski** [bitwarden/server#3116]
### Steps To Reproduce

1. Login with new device
2. Examine email notification of new device IP Address feld

### Expected Result

Real IP of logged on device is displayed

### Actual Result

IP of internal reverse proxy is displayed

### Screenshots or Videos

![image](https://github.com/bitwarden/server/assets/533249/51f66b67-3474-440c-9daf-5d8e02faa6b0)

### Additional Context

Previously we were able to get this IP reported correctly due to utilisaiton of the `X-Real-IP` header, however this appears to have been broken by the changes in https://github.com/bitwarden/server/pull/2847.

As I understand it the solution would now be to set the `KnownProxies` config option, however this is not very easily achievable in our environment as the proxies IP addresses are dynamically allocated in a dedicated 'reverse proxy' subnet and can be added to and removed dynamically.

As per my understanding of the .net side, the ideal solution would be to implement the ability to set [KnownNetworks](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.knownnetworks?view=aspnetcore-7.0)  as well as KnownProxies, so we could allocate the entire subnet as proxied.

Alternatively restore the ability to allow us to continue setting a header like `X-Real-IP` as trusted as this is a regression in functionality.

### Build Version

2023.7.0 

### Environment

Self-Hosted

### Environment Details

Docker-swarm based deployment with a caddy-based reverse proxy terminating TLS to the containers.

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Javed-Patel-wohlig** [comment]
hey, if this is not fixed yet I can contribute by fixing this.

**Comment by danpoltawski** [comment]
I don't believe its fixed

**Comment by SergeantConfused** [comment]
Hello @danpoltawski,

In testing, `X-Real-IP` header is working as expected with Server 2025.1.0 and is relaying the client's IP address. Could you please let me know if you're still experiencing this behaviour in more recent versions of the server?

Thank you in advance,

**Comment by danpoltawski** [comment]
> In testing, `X-Real-IP` header is working as expected with Server 2025.1.0 and is relaying the client's IP address. Could you please let me know if you're still experiencing this behaviour in more recent versions of the server?

I will wait for that release to be included in https://github.com/bitwarden/self-host and then re-test

**Comment by SergeantConfused** [comment]
Hi @danpoltawski,

To be clear, are you still encountering this in server 2024.11.X or 2024.12.X?

Thank you,

**Comment by danpoltawski** [comment]
Yes, still experiencing on 2024.12.1. Please see my additional context for an explanation of why we're seeing this.
